### DISCUSSION TITLE
Consuming event with [else] guard

### DISCUSSION THREAD

**Discussion by iamjuing** [StateSmith/StateSmith#134] [Ideas]
Hi, can you please confirm this is the correct/expected way of consuming event (internal transition), in case all guards are false?

![image](https://user-images.githubusercontent.com/125978139/224652798-5fc5c85d-c190-4f1b-90cf-c73112d4a491.png)

Thanks.

**Comment by adamfk** [comment]
Good question.

If you want the internal transition to happen last, you'll want to give the external transition behaviors an order so that StateSmith will check them first. https://github.com/StateSmith/StateSmith/wiki/Behaviors#ordering-a-states-behaviors

Here are some options:

![image](https://user-images.githubusercontent.com/274012/224724726-c4fb5512-468a-4f62-a281-ece4932fb0f8.png)

![image](https://user-images.githubusercontent.com/274012/224724328-f0718536-c3a7-4f45-881a-a71b0c2c34c0.png)

![image](https://user-images.githubusercontent.com/274012/224724881-1966a3aa-d80b-4b27-be6b-ca84c1619de5.png)

**Comment by iamjuing** [comment]
Understand.
Honestly, I don't know what would be the correct way to achieve this as per the UML standard.

What makes sense to me instead of using ordering extension (if I don't care about the order), would be to use `EVENT [else]` in the state as the internal transition.
Would that work?

**Comment by adamfk** [comment]
`EVENT [else]` is a good idea. I think it would work well. I think the original StateSmith actually did that. Can you please open an issue?

**Comment by iamjuing** [comment]
Sure. Thanks.
See #136
